src/undocommand.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+import Command from '../command/command.js';
+
+/**
+ * Undo command stores batches in itself and is able to and apply reverted versions of them on the document.
+ *
+ * @memberOf undo
+ */
+export default class UndoCommand extends Command {
+	constructor( editor ) {
+		super( editor );
+
+		/**
+		 * Items that are pairs of batches which are saved by the command and model selection state at the moment of saving the batch.
+		 *
+		 * @private
+		 * @member {Array.<engine.treeModel.Batch>} undo.UndoCommand#_batchStack
+		 */
+		this._batchStack = [];
+	}
+
+	/**
+	 * Stores a batch in the command. Stored batches can be then reverted.
+	 *
+	 * @param {engine.treeModel.Batch} batch Batch to add.
+	 */
+	addBatch( batch ) {
+		const selection = {
+			ranges: Array.from( this.editor.document.selection.getRanges() ),
+			isBackward: this.editor.document.selection.isBackward
+		};
+
+		this._batchStack.push( { batch, selection } );
+	}
+
+	/**
+	 * Removes all batches from the stack.
+	 */
+	clearStack() {
+		this._batchStack = [];
+	}
+
+	_checkEnabled() {
```

Comment:
Oops, it's inherited from `Command`. Hmm... so maybe, to avoid confusion, we could use `@inheritDoc` in such cases. Then it will be clear that the doc comes from the parent.